app/src/Talk/TalkController.php

```diff
@@ -475,15 +480,21 @@ public function unlinkSpeaker($eventSlug, $talkSlug, $username)
 
         $unlinkSpeakerUri = $talkUri . "/speakers/" . $userId;
 
-        try {
-            $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
-        } catch (Exception $e) {
-            $this->application->flash('error', $e->getMessage());
+        $isAdmin = $event->getCanEdit();
+        if (!$isAdmin && $talk->getStartDateTime()->format('U') > time()) {
+            try {
+                $talkApi->unlinkVerifiedSpeakerFromTalk($unlinkSpeakerUri);
+            } catch (Exception $e) {
+                $this->application->flash('error', $e->getMessage());
+                $this->application->redirect($url);
+            }
+
+            $this->application->flash('message', 'Speaker has been removed from this talk.');
+            $this->application->redirect($url);
+        } else {
```

Comment:
Since this is a one-or-the-other (true boolean) situation, I believe an if-else is the correct usage here. If the _if_ was unrelated to the outcome of the contents of the else, things would be different. In this case the _if_ is exactly about the message in the else.
I could reverse the logic if you feel that's better ?